crates/bevy_ecs/src/system/system_registry.rs

```diff
@@ -362,6 +378,8 @@ impl World {
                 ret
             });
 
+        self.resource_mut::<RunSystemStack>().remove(&id.entity);
```

Comment:
This could technically panic if a malicious system finds the `RunSystemStack` resource by ID and removes it.  It might be a little safer like

```suggestion
        if let Some(run_system_stack) = self.get_resource_mut::<RunSystemStack>() {
            run_system_stack.remove(&id.entity);
        }
```

We may even want to do the same thing for `insert`, since a user could register a malicious observer that runs during `take::<RegisteredSystem>`.

Reply:
maybe it is impossible for that to happen in between the `init_resource` and `insert`, but i will have on `remove`

Reply:
would the observer of `take` happen immediately or after the end of method? i don't know how that works when fiddling with `World`